### PR TITLE
Get Market events between blocks

### LINKED ISSUE #24: Get Market events between blocks
To avoid lost or send same message after a restart replace the event watcher to pull events between fromBlock: `lastUsedBlock` and toBlock: `nodeLastBlock-5`:
- lastUsedBlock: the value of `nodeLastBlock-5` from the last loop execution.
- nodeLastBlock-5: the last block number of the node minus 5 blocks to avoid reorgs.

### PR DESCRIPTION
**Pull request by girianshido**
Add `watchFPMMCreationEvent` function to get past events between
`fromBlock` to `toBlock` block number.

Add a new job for `watchNewMarketsEvent` from started block `fromBlock`.

Resolves: #24

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -23,11 +23,18 @@ const startMessage = `Conditional Tokens bot \`${version}\` was started.`;
 pushSlackMessage(startMessage);
 console.log(startMessage);
 
+const jobTime = process.env.JOB_GET_TRADE_MINUTES ? process.env.JOB_GET_TRADE_MINUTES : 5;
```
Comment (mariano-aguero) [inline src/index.js:27]: You can create a CONFIG file, or maybe a constants file and then import it, so you don't to initialize this variable here

File: src/events/marketEvents.js [hunk 2]
```diff
@@ -7,69 +7,79 @@ const { getTokenName, getTokenSymbol } = require('../services/contractERC20');
...
-    });
+            })();
+        });
+    })
+}
+
+/**
+ * Watch `FixedProductMarketMakerCreation` events from a `FixedProductionMarketMakerFactory` contract.
+ */
+module.exports.watchNewMarketsEvent = async (fromBlock) => {
```
Comment (mariano-aguero) [inline src/events/marketEvents.js:77]: Maybe you can move this function to some utils file ?

File: src/events/marketEvents.js [hunk 3]
```diff
@@ -7,69 +7,79 @@ const { getTokenName, getTokenSymbol } = require('../services/contractERC20');
...
-                                console.log(event.returnValues.conditionIds[0] + ':\n' + message.join('\n') + '\n\n');
-                            });
+        fromBlock,
+        toBlock,
+    }, (error, events) => {
+        events.forEach(event => {
+            (async () => {
+                const message = new Array('<!here>', '*New market created!* :tada:');
+                //TODO support conditional markets
+                Promise.all([
```
Comment (mariano-aguero) [inline src/events/marketEvents.js:32]: You can await this Promise All, for example
```
 let [t1, t2, ...] = await Promise.all([
    task1(...),
    task2(...),
    ....
  ])
```

File: src/events/marketEvents.js [hunk 4]
```diff
@@ -7,69 +7,79 @@ const { getTokenName, getTokenSymbol } = require('../services/contractERC20');
...
+                    getCondition(event.returnValues.conditionIds[0]),
+                    web3.eth.getTransaction(event.transactionHash),
+                    getTokenName(web3, event.returnValues.collateralToken), 
+                    getTokenSymbol(web3, event.returnValues.collateralToken), 
+                ])
+                    .then(([conditions, transaction, tokenName, tokenSymbol]) => {
+                        conditions.forEach(condition => {
```
Comment (mariano-aguero) [inline src/events/marketEvents.js:39]: If the purpose of this , is to generate an array of messages, you can iterate over a map with async / await, so you avoid to use a lot of `then`

**Review by mariano-aguero — CHANGES_REQUESTED** [review]
Great Job! I added some comments

File: src/events/marketEvents.js (hunk 4 above)
Comment (girianshido) [reply src/events/marketEvents.js:39]: Thanks Mariano! I will add these suggestion in other new PR

File: src/index.js (hunk 1 above)
Comment (girianshido) [reply src/index.js:27]: Created issue

File: src/events/marketEvents.js (hunk 2 above)
Comment (girianshido) [reply src/events/marketEvents.js:77]: https://github.com/gnosis/conditional-tokens-bot/issues/29

File: src/index.js (hunk 1 above)
Comment (girianshido) [reply src/index.js:27]: https://github.com/gnosis/conditional-tokens-bot/issues/28

File: src/events/marketEvents.js (hunk 3 above)
Comment (girianshido) [reply src/events/marketEvents.js:32]: https://github.com/gnosis/conditional-tokens-bot/issues/30

File: src/events/marketEvents.js (hunk 4 above)
Comment (girianshido) [reply src/events/marketEvents.js:39]: https://github.com/gnosis/conditional-tokens-bot/issues/30

**Pull request merged**